### PR TITLE
Kotlin conversion: UriUtil

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T272618

### CODE REVIEW

File: app/src/main/java/org/wikipedia/util/UriUtil.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    @JvmStatic
+    fun getUrlWithProvenance(context: Context, title: PageTitle,
+                             @StringRes provId: Int): String {
+        return title.uri + "?wprov=" + context.getString(provId)
+    }
+
+    @JvmStatic
+    fun getFilenameFromUploadUrl(url: String): String {
+        val splitArray = url.split("/".toRegex()).toTypedArray()
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/UriUtil.kt:107]: Is the `toTypedArray()` necessary? Looks like we can rewrite this method.

File: app/src/main/java/org/wikipedia/util/UriUtil.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    @JvmStatic
+    fun decodeURL(url: String): String {
+        return try {
+            // Force decoding of plus sign, since the built-in decode() function will replace
+            // plus sign with space.
+            URLDecoder.decode(url.replace("+", "%2B"), "UTF-8")
+        } catch (e: IllegalArgumentException) {
+            // Swallow IllegalArgumentException (can happen with malformed encoding), and just
+            // return the original string.
+            d("URL decoding failed. String was: $url")
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/UriUtil.kt:34]: Can we have a standard `L.d()` at this line?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/util/UriUtil.kt (hunk 1 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/util/UriUtil.kt:107]: @dbrant @cooltey Wondering if we sshould filter out spaces here too?  We might need to remove the test for empty string then : testFilenameFromUploadUrl -> assertThat(UriUtil.getFilenameFromUploadUrl(""), is(""));

File: app/src/main/java/org/wikipedia/util/UriUtil.kt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        return removeFragment(removeLinkPrefix(url)).replace("_", " ")
+    }
+
+    /** Get language variant code from a Uri, e.g. "zh.*", otherwise returns empty string.  */
+    @JvmStatic
+    fun getLanguageVariantFromUri(uri: Uri): String {
+        if (uri.path.isNullOrEmpty()) {
+            return ""
+        }
+        val parts = StringUtils.defaultString(uri.path).split('/')
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/UriUtil.kt:125]: Since we already have the `isNullOrEmpty()`, do we still need this `StringUtils.defaultString()` instead of using `uri.path!!`?
